core/src/module/usModule.cpp

```diff
@@ -119,15 +121,37 @@ bool Module::IsLoaded() const
 
 void Module::Start()
 {
-
+  MutexLockingStrategy::Lock(this->d);
```

Comment:
After looking at the problem, I don't see a good solution, that doesn't hold a lock while calling into client code, which doesn't require a significant change.

I removed the use of ````MutexLockingStrategy```` and replaced it with ````MutexLock```` and ````Mutex```` since its simpler.

I will "opt out" and add information to [the multithreading issue](https://github.com/saschazelzer/CppMicroServices/issues/32) so we can re-visit it.